src/Toolkit/PubSub/V1/PublisherApi.php

```diff
@@ -0,0 +1,673 @@
+     *                             `"projects/{project}/topics/{topic}"`. `{topic}` must start with a letter,
+     *                             and contain only letters (`[A-Za-z]`), numbers (`[0-9]`), dashes (`-`),
+     *                             underscores (`_`), periods (`.`), tildes (`~`), plus (`+`) or percent
+     *                             signs (`%`). It must be between 3 and 255 characters in length, and it
+     *                             must not start with `"goog"`.
+     * @param array  $optionalArgs {
+     *                             Optional. There are no optional parameters for this method yet;
+     *                             this $optionalArgs parameter reserves a spot for future ones.
+     *                             }
+     * @param array  $callSettings {
+     *                             Optional.
+     *
+     *    @var Google\GAX\RetrySettings $retrySettings
+     *          Retry settings to use for this call. If present, then
+     *          $timeout is ignored.
+     *    @var int $timeoutMillis
+     *          Timeout to use for this call. Only used if $retrySettings
+     *          is not set.
+     * }
+     *
+     * @return google\pubsub\v1\Topic
+     *
+     * @throws Google\GAX\ApiException if the remote call fails
+     */
+    public function createTopic($name, $optionalArgs = [], $callSettings = [])
+    {
+        $request = new Topic();
+        $request->setName($name);
+
+        $mergedSettings = $this->defaultCallSettings['createTopic']->merge(
+            new CallSettings($callSettings)
+        );
+        $callable = ApiCallable::createApiCall(
+            $this->stub,
+            'CreateTopic',
+            $mergedSettings,
+            $this->descriptors['createTopic']
+        );
+
+        return $callable(
+            $request,
+            [],
+            ['call_credentials_callback' => $this->createCredentialsCallback()]);
+    }
+
+    /**
+     * Adds one or more messages to the topic. Returns `NOT_FOUND` if the topic
+     * does not exist. The message payload must not be empty; it must contain
+     *  either a non-empty data field, or at least one attribute.
+     *
+     * Sample code:
+     * <pre><code>
+     * try {
+     *     $publisherApi = new PublisherApi();
+     *     $formattedTopic = PublisherApi::formatTopicName("[PROJECT]", "[TOPIC]");
+     *     $data = "";
+     *     $messagesElement = new PubsubMessage();
+     *     $messagesElement->setData($data);
+     *     $messages = [$messagesElement];
+     *     $response = $publisherApi->publish($formattedTopic, $messages);
+     * } finally {
+     *     if (isset($publisherApi)) {
+     *         $publisherApi->close();
+     *     }
+     * }
+     * </code></pre>
+     *
+     * @param string $topic        The messages in the request will be published on this topic.
+     * @param array  $messages     The messages to publish.
```

Comment:
When we poked around at existing documentation conventions, it just looked like `array` was the convention; we can easily switched to your proposed form - it is definitely more informative.